core/trino-main/src/main/java/io/trino/sql/gen/JoinCompiler.java

```diff
@@ -168,14 +179,17 @@ private LookupSourceSupplierFactory internalCompileLookupSourceFactory(List<Type
     {
         Class<? extends PagesHashStrategy> pagesHashStrategyClass = internalCompileHashStrategy(types, outputChannels, joinChannels, sortChannel);
 
+        boolean singleChannelBigint = enableSingleChannelBigintLookupSource &&
+                joinChannels.size() == 1 &&
+                types.get(getOnlyElement(joinChannels)) == BIGINT;
         Class<? extends LookupSourceSupplier> joinHashSupplierClass = IsolatedClass.isolateClass(
                 new DynamicClassLoader(getClass().getClassLoader()),
                 LookupSourceSupplier.class,
                 JoinHashSupplier.class,
                 JoinHash.class,
-                DefaultPagesHash.class);
-
-        return new LookupSourceSupplierFactory(joinHashSupplierClass, new PagesHashStrategyFactory(pagesHashStrategyClass));
+                singleChannelBigint ? BigintPagesHash.class : DefaultPagesHash.class);
```

Comment:
Do we need to isolate bigint hash? Does it call any interface with multiple implementations?

Reply:
Isn't the fact that `PagesHash` is an interface relevant? TBH I don't know much about class isolation. It was isolated before so I just added the new class